### ISSUE TITLE
All conversations are accessible using a link without even having an account

### ISSUE THREAD

**Issue by softwareclass2** [nextcloud/spreed#7474]
## Steps to reproduce
All private conversation or group conversations are accessible via a link (https://[domain.com]/index.php/call/[id]). There is no way to disable this feature. In case someone find the link they can access the conversation without even having an account. In case there is a group conversation and a member leave the conversation or the account gets terminated, they can indefinitely access the future conversations. 

### Expected behaviour
Direct link should be disabled by default for security reasons and should be changeable even if they were activated by users (in this case they are activated by default). Direct link should not be accessible to public unless it was configured for public conversations 

### Actual behaviour
Tell us what happens instead

## Browser

**Microphone available:** yes/no

**Camera available:** yes/no

**Operating system:** Windows/Ubuntu/...

**Browser name:** Firefox/Chrome/...

**Browser version:** 50.1/55/...

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Talk app

**Talk app version:** (see apps admin page: `/index.php/settings/apps`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk`)

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:**

**Nextcloud Version:** (see admin page)

**List of activated apps:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

**Comment by nickvergessen** [comment]
Just uncheck the "Allow guests" in the conversation settings?

**Comment by softwareclass2** [comment]
> Just uncheck the "Allow guests" in the conversation settings?

There is no such an option, but even if there was such an option, then allowing a guest should not be active by default. You put many people at risk while they trust your platform with their private conversations.

**Comment by softwareclass2** [comment]
I personally switched from WhatsApp to NextCloud Talk for privacy. Now, I realize that the most important thing that I never had was privacy.

**Comment by nickvergessen** [comment]
Conversations are not accessible to unregistered users by default.

**Comment by softwareclass2** [comment]
> Conversations are not accessible to unregistered users by default.

Conversation can be accessible to anyone using a direct URL and **there is no way to stop this**. 
If there is anyway to stop accessing the conversation via direct URLs and to people who are not registered or to those who are not part of a conversation anymore, then please let me know so I will change that. I checked all the options and there is no way I can stop direct access to the conversations. 
I also installed this server myself and did not make any changes to give direct access to conversations. Nevertheless, even if I were not the server admin and even as a regular user I have to be able to block direct access to my personal conversation. This is security and privacy 101 principle.

**Comment by pierreozoux** [comment]
You should disclose security issue on hackerone:

Like this one:
https://hackerone.com/reports/428010

You might get a bounty ;)

(And it is more responsible for the project and people using it)

**Comment by tcitworld** [comment]
So you don't have this option in the conversation settings?
![image](https://user-images.githubusercontent.com/2197836/173176498-cc2e1bdf-3e21-497d-bbdc-9469a11241e3.png)

**Comment by nickvergessen** [comment]
The setting outlined by tcit is what you are looking for.
You can also see whether a conversation is open or not by the icon of the conversation in the left sidebar:

* Chain ⛓️ icon: public conversation accesible by guests
* People 👥 icon: private conversation not accessible by guests (they can still put the link into their browser, but it will only show the login form)

**Comment by softwareclass2** [comment]
> So you don't have this option in the conversation settings?

I updated the server and now I can see this option. I disabled it and it seems that now it works; yet I cannot understand how it was active in the first place. Maybe if someone mistakenly click on the link it gets activated for all users. 
Also, it is only accessible via browser I can't see it with my android phone.

Nothing important, but we had a family chat group for my mom and my sisters. One of the sisters left the group but she could manage to get access to all the conversations even after she was not a registered user and we did not even know that.
